### PR TITLE
Feature/jenkins 36973 favorite stop build

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- See [JENKINS-36973](https://issues.jenkins-ci.org/browse/JENKINS-36973).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
  - Ensure some favorites are added to the Dashboard, ideally a freestyle, pipeline and multibranch pipeline branch with gitflow-style name.
  - Run each of the builds and ensure that the "queued" and "started" toasts display in the UI. Verify that the "Open" link works correctly for each of the "started" toasts.
  - While a build is running click the "Stop" button in the card.
    - Verify the build stops and the card updates. In most cases this should happen very quickly (< 1s) but for some involved pipelines it could take up to 10s.
    - Verify the "stopping" and "stopped" toasts display correctly.
    - Verify the "stop" button becomes disabled while the stop is in the progress.
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

File: blueocean-personalization/src/main/less/icons/stop-white.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (i386) [inline blueocean-personalization/src/main/less/icons/stop-white.svg]: Might want to run this through a SVG optimiser :)

Comment (cliffmeyers) [inline blueocean-personalization/src/main/less/icons/stop-white.svg]: Fixed in 6d763a6

**Comment by cliffmeyers** [comment]
@michaelneale was wondering if you could give this a try now. I did not implement the functionality that re-enables the stop button after 10s, only because in the vast majority of my testing the builds seem to stop within 1s, if not well under 10s. If you don't mind playing with it and letting me know whether or not you feel we need that behavior, that'd be great. Can certainly implement it, but since this code isn't being shared it might be a small dup of effort. I do plan to impl the 10s thing for the work in JENKINS-36687 and JENKINS-36974.

**Comment by i386** [comment]
Looks good from my Pov. Will wait for someone to give you a code review.

**Comment by cliffmeyers** [comment]
ATH passed locally. Trying to a run in CI now too.

**Comment by i386** [comment]
@cliffmeyers hold on merging until master is working again

**Comment by cliffmeyers** [comment]
@i386 sure thing

**Comment by michaelneale** [comment]
sorry @cliffmeyers didn't get a chance. Try again after updating to master, ATH locally etc etc..

**Comment by cliffmeyers** [comment]
@michaelneale this one is still conflict-free and hasn't changed since local ATH passed for me. Per @i386 we're okay with the lack of the 10s re-enable stop button thing, so I'm going to mark this as code complete.

**Comment by michaelneale** [comment]
Looks good, seems to work well with those combinations. One small :ant: is that I didn't see it show up the "stopped" toast, only ever "stopping", but that didn't bother me (ends up being too many toasts anyway...)

I wasn't able to get a pipeline that took a while to stop, so hard to tell the hard stop behavior. 

I am inlined to 🐝  this but I would like it updated from master first, to check that the scenarios still hold AND the ATH runs locally at least for you. Nice work!

**Comment by michaelneale** [comment]
ping @cliffmeyers this is still ready - or you waiting for JWT to land this? (ideally would be udpated to master)

**Comment by cliffmeyers** [comment]
@michaelneale yes, I was waiting for JWT to land. If I can merge this quickly and easily later today I will. If it's going to be a headache, then I may close this PR and re-open a new one that uses the new API's from #448.

**Comment by cliffmeyers** [comment]
Local ATH passed.

`2016-09-07 12:11:48,087 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - OK. 407  total assertions passed. (4m 22s)`

**Comment by cliffmeyers** [comment]
So I'm going to merge this now to get these features in. I have a ticket where I am streamlining the toast behavior to show only "STARTED" and "STOPPING" toasts (eliminating the spammy "QUEUED" and "STOPPED" toasts) for runs and stops in `dashboard`. In that ticket, I will also rework some of the code added here to use the same unified code which will clean this all up quite a bit.

https://issues.jenkins-ci.org/browse/JENKINS-37425

Right now I'd prefer not to make the change before JW but if we really want that new behavior, I can put it in as part of separate PR.

**Pull request merged**